### ISSUE TITLE
How to select best number of iterations

### ISSUE THREAD

**Issue by ThorvaldAagaard** [lorserker/ben#12]
I am training very specific sets with 10.000 input, and when using 
```
bidding_nn.py
```
it is hardcoded to run 1.000.000 iterations

Is 1.000.000 iterations needed for smaller input sets?

If not, is there any recommendations for selecting the optimal number of iterations.

If it makes sense to change the number  of iterations, it might be an idea to be able to set the number of iterations, when calling bidding_nn.py

**Comment by lorserker** [comment]
with fewer boards it's better to run fewer iterations. there is no rule of thumb how many, you can look at the learning curve and at metrics computed on a validation set.

**Comment by ThorvaldAagaard** [comment]
Can I find the explanations of the metrics and on the output from bidding_nn.py ?

**Comment by lorserker** [comment]
@ThorvaldAagaard  the output of the nn is a [softmax](https://deepai.org/machine-learning-glossary-and-terms/softmax-layer)
the numbers that the script prints as it progresses is the loss function, which is in this case [cross entropy](https://en.wikipedia.org/wiki/Cross_entropy) and computed by this [function in tensorflow](https://www.tensorflow.org/api_docs/python/tf/compat/v1/losses/softmax_cross_entropy)
intuitively, it's a measure of how well the model fits the training data (the lower the number the better)

**Comment by lorserker** [comment]
closing
